src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Custom/Native/Call/NativeCallTracerCallFrame.cs

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using Nethermind.Core;
+using Nethermind.Int256;
+
+namespace Nethermind.Evm.Tracing.GethStyle.Custom.Native.Call;
+
+[JsonConverter(typeof(NativeCallTracerCallFrameConverter))]
+public class NativeCallTracerCallFrame
+{
+    public Instruction Type { get; set; }
+
+    public Address? From { get; set; }
+
+    public long Gas { get; set; }
+
+    public long GasUsed { get; set; }
+
+    public Address? To { get; set; }
+
+    public byte[]? Input { get; set; }
+
+    public byte[]? Output { get; set; }
+
+    public string? Error { get; set; }
+
+    public string? RevertReason { get; set; }
+
+    public List<NativeCallTracerCallFrame> Calls { get; set; }
+
+    public List<NativeCallTracerLogEntry>? Logs { get; set; }
```

Comment:
Potentially `Input`, `Output`, `Calls` and `Logs` could be replaced with `ArrayPoolList` if they would be then later correctly disposed. This would reduce memory allocations.

Reply:
I attempted to spike out an approach for converting these fields to be of type `ArrayPoolList` instead, however I was having difficulty finding a way to only dispose of them after the trace object is serialized to JSON. I tried using the `Dispose` method from the `TxTracer` abstract class, however that would dispose of the lists before they were able to be serialized.

I looked around our other pre-existing traces (`GethLikeTxTrace`/`ParityLikeTxTrace`) to try to find an example of serializing `ArrayPoolList` objects, however it seems like they just use the built in list type (`[]`) or the `List` object type as well.

I agree that it would be nice to find a way to use `ArrayPoolList` for list type trace fields to reduce memory allocations, however it might make sense to move this into a separate issue due to the added complexity with disposing after serialization and also so we can optimize the existing opcode traces that store list type fields as well.

Here was my spiked approach for reference: [SPIKE](https://github.com/NethermindEth/nethermind/commit/59761b36b1449ff9498bbdb8db98a04488113174)

Reply:
Our JSON RPC stack is built this way that if the serialized object is `IDisposable` then it's disposed will be called. as you can see `GethLikeTxTrace` is implementing disposable and allowing to pass custom internal one to get advanced logic. So you need to pass some custom disposable there that will dispose your `ArraPoolLists`

Reply:
Ah ok that makes sense, thanks for the clarification. I'll make these updates to dispose of the `ArrayPoolList` fields at the call frame level instead of at the tracer level so that they are disposed after serialization.